pandas/core/frame.py

```diff
@@ -6765,7 +6765,7 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
                 other.values.reshape((1, len(other))),
                 index=index,
                 columns=combined_columns,
-            )
+            ).astype(other.dtype)
```

Comment:
this is not great way to fix this
the construction needs to change

Reply:
@jreback thanks for your quick review. I tried passing the `dtype` argument directly, but then ran into #30280.

However, I've been able to get around the problem by rewriting the code a bit (as I understand, chaining methods is now the preferred way of using Pandas). The tests pass.

What I don't (yet) understand is the failing CI build:
```
error: Module has no attribute "FixedOffset"
```
Other parts of the code seem to be using
```
import pytz
tz=pytz.FixedOffset
```
without any issues, but I'll look into it